unittests/test-contracts/savanna/common/bitset.hpp

```diff
@@ -0,0 +1,82 @@
+using namespace eosio;
+
+namespace savanna {
+
+    class bitset {
+        using word_t = uint8_t;
+        static constexpr size_t num_bits = sizeof(word_t) * 8;
+        static size_t round_up(size_t sz) { return (sz + num_bits - 1) / num_bits; }
+
+    public:
+
+        // compare two bitsets
+        static std::pair<savanna::bitset, savanna::bitset> compare(const savanna::bitset& bs1, const savanna::bitset& bs2) {
+            check(bs1.size() == bs2.size(), "bitsets must be of the same size");
+
+            size_t size = bs1.size();
+            std::vector<word_t> intersection_data(bs1.data.size());
+            std::vector<word_t> symmetric_difference_data(bs1.data.size());
+
+            for (size_t i = 0; i < bs1.data.size(); ++i) {
+                intersection_data[i] = bs1.data[i] & bs2.data[i];
+                symmetric_difference_data[i] = bs1.data[i] ^ bs2.data[i];
+            }
+
+            savanna::bitset intersection(size, intersection_data);
+            savanna::bitset symmetric_difference(size, symmetric_difference_data);
+
+            return {intersection, symmetric_difference};
+        }
+
+        bitset(word_t size) 
+            : data(round_up(size), 0) {}
+        
+        bitset(word_t size, const std::vector<word_t> raw_bitset)
+           : data(raw_bitset) {
+               check(raw_bitset.size() == round_up(size), "invalid raw bitset size");
+           }
+
+        // Set a bit to 1
+        void set(word_t index) {

```

Comment:
Please do same change for `clear()` and `test()` and `check_bounds()`
```suggestion
        void set(size_t index) {
```

Reply:
Please do same change for clear() and test() and check_bounds()

Reply:
Oops, I missed the comment above, sorry... I fixed it now.

Reply:
Perfect, thanks!